src/game_api/entity.cpp

```diff
@@ -402,22 +402,60 @@ void Entity::destroy()
     delete this; // TODO
 }
 
-std::tuple<float, float, int> get_position(uint32_t id)
+std::tuple<float, float, int> get_position(uint32_t uid)
 {
-    Entity* ent = get_entity_ptr(id);
+    Entity* ent = get_entity_ptr(uid);
     if (ent)
         return std::make_tuple(ent->position().first, ent->position().second, ent->layer);
     return {0.0f, 0.0f, 0};
 }
 
-std::tuple<float, float, int> get_render_position(uint32_t id)
+std::tuple<float, float, int> get_render_position(uint32_t uid)
 {
-    Entity* ent = get_entity_ptr(id);
+    Entity* ent = get_entity_ptr(uid);
     if (ent)
         return std::make_tuple(ent->position_render().first, ent->position_render().second, ent->layer);
     return {0.0f, 0.0f, 0};
 }
 
+std::tuple<float, float> get_velocity(uint32_t uid)
+{
+    if (Entity* ent = get_entity_ptr(uid))
+    {
+        float vx{0.0f};
+        float vy{0.0f};
+        if (ent->type->search_flags & 0b1111) // PLAYER | MOUNT | MONSTER | ITEM
+        {
+            Movable* mov = ent->as<Movable>();
+            vx = mov->velocityx;
+            vy = mov->velocityy;
+        }
+        if (ent->overlay)

```

Comment:
shouldn't this be a loop to determinate top entity?
like when you on a mount and elevator, it will return 0. And it can goo even further if you check speed of jetpack

also fx and activefloor are also movable

Reply:
See line 435

Reply:
Will fix fx and activefloor, thanks for that  👍